draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -848,6 +848,152 @@ Content-Type: text/plain
 </artwork>
 </section>
 
+<section title="Origin" anchor="http.origin">
+<t>
+   The origin for a given URI is the triple of scheme, host, and port
+   after normalizing the scheme and host to lowercase and normalizing the
+   port to remove any leading zeros. If port is elided from the URI, the
+   default port for that scheme is used. For example, the URI
+</t>
+<sourcecode>
+   https://Example.Com/happy.js
+</sourcecode>
+<t>
+   would have the origin
+</t>
+<sourcecode>
+   { "https", "example.com", "443"}
+</sourcecode>
+<t>
+   which can also be described as the normalized URI prefix with port always
+   present:
+</t>
+<sourcecode>
+   https://example.com:443
```

Comment:
Is the intent here to supplant RFC 6454 rather than reference it?

Reply:
My intent is to supply the normative bits specifically for HTTP origin, which a later update to 6454 might use as part of its larger description of web origins for HTML scripting. I move the informational reference to 6454 up to this section. It's a little confusing because the web origin concept is based on the HTTP origin server definitions but applied to many other references as well.